src/NetTopologySuite.IO.GeoJSON4STJ/Converters/StjFeatureConverter.cs

```diff
@@ -107,14 +107,14 @@ public override IFeature Read(ref Utf8JsonReader reader, Type objectType, JsonSe
                                 break;
 
                             case JsonTokenType.Number:
-                                throw new NotSupportedException("Number value cannot be boxed as a decimal: " + reader.GetString());
+                                throw new ParseException(new NotSupportedException("Number value cannot be boxed as a decimal: " + reader.GetString()));
 
                             case JsonTokenType.String:
                                 feature.Id = reader.GetString();
                                 break;
 
                             default:
-                                throw new JsonException("A GeoJSON Feature's \"id\", if specified, must be either a JSON string or number, per RFC7946 section 3.2");
+                                throw new ParseException(new JsonException("A GeoJSON Feature's \"id\", if specified, must be either a JSON string or number, per RFC7946 section 3.2"));
```

Comment:
the inner exceptions don't really need to be those types
```suggestion
                                throw new ParseException("A GeoJSON Feature's \"id\", if specified, must be either a JSON string or number, per RFC7946 section 3.2");
```

Reply:
dito